browser/js/dashboard/gridbox/dashboard.factory.js

```diff
@@ -7,7 +7,15 @@ app.factory('DashboardFactory', function($http, $q, GeneratorFactory){
         }
         return $http.get(link)
         .then(getData)
-        .catch(() => [])
+        .then(function(data){
+            let x = obj.findDataToGraph(data);
+            if (x) {
+                return x;
+            }else{
+                return [data]
```

Comment:
Yeah, it looks like this whole logic is repeated a couple times, but could instead be moved to `.findDataToGraph`.